Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -228,11 +217,34 @@ public override void Initialize()
             handTrackingInputEventData = new HandTrackingInputEventData(EventSystem.current);
         }
 
+        private void InstantiateGazeProvider(MixedRealityPointerProfile pointerProfile)
+        {
+            if (pointerProfile?.GazeProviderType?.Type != null)
+            {
+                GazeProvider = CameraCache.Main.gameObject.EnsureComponent(pointerProfile.GazeProviderType.Type) as IMixedRealityGazeProvider;
+                GazeProvider.GazeCursorPrefab = pointerProfile.GazeCursorPrefab;
+                // Current implementation implements both provider types in one concrete class.
+                EyeGazeProvider = GazeProvider as IMixedRealityEyeGazeProvider;
```

Comment:
Worth adding a check if EyeGazeProvider null and output warning? 

This seems like a hidden thing that the GazeProvider needs to implement both interfaces

Reply:
EyeGazeProvider and GazeProvider are the same thing, but i do agree.